Pcap++/src/PcapLiveDevice.cpp

```diff
@@ -105,6 +105,66 @@ PcapLiveDevice::PcapLiveDevice(pcap_if_t* pInterface, bool calculateMTU, bool ca
 		LOG_DEBUG("   Default Gateway: " << m_DefaultGateway);
 	}
 
+	//copy interface info
+	ifaceInfo = new pcap_if_t;
+	if(pInterface->name)
+	{
+		ifaceInfo->name = new char[strlen(pInterface->name) + 1];
+		strcpy(ifaceInfo->name, pInterface->name);
+	}
+	else
+		ifaceInfo->name = NULL;
+	if(pInterface->description)
+	{
+		ifaceInfo->description = new char[strlen(pInterface->description) + 1];
+		strcpy(ifaceInfo->description, pInterface->description);
+	}
+	else
+		ifaceInfo->description = NULL;
+
+	if(pInterface->addresses)
+	{
+		ifaceInfo->addresses = new pcap_addr;
+		ifaceInfo->addresses->next = NULL;
+
+		if(pInterface->addresses->addr)
+		{
+			ifaceInfo->addresses->addr = new sockaddr;
+			memcpy(ifaceInfo->addresses->addr, pInterface->addresses->addr, sizeof(sockaddr));
+		}
+		else
+			ifaceInfo->addresses->addr = NULL;
+
+		if(pInterface->addresses->netmask)
+		{
+			ifaceInfo->addresses->netmask = new sockaddr;
+			memcpy(ifaceInfo->addresses->netmask, pInterface->addresses->netmask, sizeof(sockaddr));
+		}
+		else
+			ifaceInfo->addresses->netmask = NULL;
+
+		if(pInterface->addresses->broadaddr)
+		{
+			ifaceInfo->addresses->broadaddr = new sockaddr;
+			memcpy(ifaceInfo->addresses->broadaddr, pInterface->addresses->broadaddr, sizeof(sockaddr));
+		}
+		else
+			ifaceInfo->addresses->broadaddr = NULL;
+
+		if(pInterface->addresses->dstaddr)
+		{
+			ifaceInfo->addresses->dstaddr = new sockaddr;
+			memcpy(ifaceInfo->addresses->dstaddr, pInterface->addresses->dstaddr, sizeof(sockaddr));
+		}
+		else
+			ifaceInfo->addresses->dstaddr = NULL;
+	}
+	else
+		ifaceInfo->addresses = NULL;
+
+	ifaceInfo->next = NULL;
+	ifaceInfo->flags = pInterface->flags;

```

Comment:
Yeah it will works but will it really "clone"? For example during runtime, for some reason one of the addresses changed. Then if we call the clone() it will construct with new values. Same question valid for PcapLiveDeviceList too. Currently at the PR PcapDeviceList constructs the new values. I was planning to change that to the copy too but which one will be better? 

For tests I will try to add asap but it can take some time. Additionally I should prepare a windows environment for manual testing. I don't know it really works for reading multiple instances of same device. If it works maybe I can add an example code too.

Reply:
I agree it might not be a perfect "clone", but in most cases devices don't change and even if they do, the clone should get the new params. If `clone()` can't find a device with the same name as the current device, it can return `NULL`.

Regarding tests - as mentioned in [this thread](https://groups.google.com/g/pcapplusplus-support/c/cqyiiZaMCyk) parallel packet capture from the same device might not be possible on all platforms, but "cloning" the device should be possible regardless capture is possible or not. It'd be great if you can test it on Windows, but even if you can't - for this PR it's enough that we can test the clone rather than capturing packets